monteCarloTreeSearch.c

```diff
@@ -0,0 +1,383 @@
+				if(numOfActions != 0)
+				{
+					//Randomly select a move to make
+					randomValue = numOfActions / 2;
+					randomValue = rand() % randomValue ;  
+					boardMove[0] = allActions[randomValue * 2];
+					boardMove[1] = allActions[randomValue * 2 + 1]; 
+				}
+				else
+				{
+					boardMove[0] = -1;
+					boardMove[1] = -1;
+				}
+				//player cannot make a move
+				if(boardMove[0] == -1 && boardMove[1] == -1)
+				{
+
+					//Check if other player can move
+					//fix;	flipMove(boardState->whoseMove); 
+					selected.children[j]->board.whoseMove = (selected.children[j]->board.whoseMove == 1) ? 2 : 1;
+
+					numOfActions = 0;
+					//get board structure from the node call getmoves and for loop with while loop inside
+					//placePiece()
+					//get the move to be made and play the game until the player wins or loses
+					getMoves(&boardState, allActions);
+
+					for(i = 0; i < 200; i++)
+					{
+						if(allActions[i] == -1)
+						{
+							break;
+						}
+
+						numOfActions++;
+					}
+					//make sure that there are still actions to be taken before the game ends
+					if(numOfActions != 0)
+					{
+						//Randomly select a move to make
+						randomValue = numOfActions / 2;
+						randomValue = rand() % randomValue ;  
+						boardMove[0] = allActions[randomValue * 2];
+						boardMove[1] = allActions[randomValue * 2 + 1]; 
+					}
+					else
+					{
+						boardMove[0] = -1;
+						boardMove[1] = -1;
+					}
+
+					//determine the winner because other player cannot make a move either
+					if(boardMove[0] == -1 && boardMove[1] == -1)
+					{
+						gameContinues = -10;
+						winner = determineWinner(&boardState);
+						if(winner == 0)
+						{
+							selected.children[j]->numSimulations += 1;
+
+						}
+						else if(winner == 1)
+						{
+							if(selected.children[j]->board.whoseMove == 1)
+							{
+								selected.children[j]->numWins += 1;
+							}
+							selected.children[j]->numSimulations += 1;
+
+						}	
+						else if(winner == 2)
+						{
+							if(selected.children[j]->board.whoseMove == 2)
+							{
+								selected.children[j]->numWins += 1;
+							}
+							selected.children[j]->numSimulations += 1;
+
+						}
+					}	
+					//Continue the game
+					else
+					{
+						placePiece(&boardState, boardMove);
+					}
+				}
+				else
+				{	
+					placePiece(&boardState, boardMove);
+				}
+
+			}
+		}
+		printf("After Simulation\n");
+
+		//4. Back-propagate results up to the root node once you get the results of your simulation.
+		//This just involves recording progress in the numWins and numSimulations members of nodes and 
+		//following parent up the tree till you hit NULL.
+	
+		printf("Back Propagation\n");
+
+		numWins = 0;
+		numSims = 0;	
+		for(i = 0; i < selected.numChildren; i++)
+		{
+		 	
+	 		numWins += selected.children[i]->numWins;
+		 	numSims += selected.children[i]->numSimulations;
+		}
+		parentNode = selected;
+		
+		while(parentNode.parent !=NULL)
```

Comment:
I think this may cause the root node to not be updated. Change the condition to `parentNode != NULL`.